integration-tests/src/test/java/oracle/weblogic/kubernetes/ItServerStartPolicy.java

```diff
@@ -1130,6 +1133,133 @@ public void testRestartingMSWithExplicitServerStartStateWhileVaryingReplicaCount
+
+    // restore test env
+    result =  assertDoesNotThrow(() ->
+        executeLifecycleScript(STOP_SERVER_SCRIPT, SERVER_LIFECYCLE, configServerName),
+        String.format("Failed to run %s", STOP_SERVER_SCRIPT));
+    checkPodDeleted(configServerPodName, domainUid, domainNamespace);
+    logger.info("managed server " + configServerPodName + " stopped successfully.");
+  }
+
+  /**
+   * Scale the dynamic cluster using the sample script scaleCluster.sh script
+   * Verify that server(s) in the dynamic cluster are scaled up and in RUNNING state.
+   * Verify that server(s) in the configured cluster are not affected.
+   * Restore the env using the sample script stopServer.sh.
+   */
+  @Order(20)
+  @Test
+  @DisplayName("Scale the dynamic cluster with scaleCluster.sh script")
+  public void testDynamicClusterScale() {
+    int newReplicaCount = 2;
+    String dynamicServerName = "managed-server" + newReplicaCount;
+    String dynamicServerPodName = domainUid + "-" + dynamicServerName;
+    String configServerPodName = domainUid + "-config-cluster-server" + newReplicaCount;
+
+    // make sure the server-to-be-test doesn't exist
+    restoreEnv();
+    checkPodDoesNotExist(configServerPodName, domainUid, domainNamespace);
+    checkPodDoesNotExist(dynamicServerPodName, domainUid, domainNamespace);
+
+    // use scaleCluster.sh to scale a dynamic cluster
+    logger.info("Scale dynamic cluster using the script scaleCluster.sh");
+    String result =  assertDoesNotThrow(() ->
+        executeLifecycleScript(SCALE_CLUSTER_SCRIPT, CLUSTER_LIFECYCLE, CLUSTER_1, " -r " + newReplicaCount, false),
+        String.format("Failed to run %s", SCALE_CLUSTER_SCRIPT));
+
+    checkPodReadyAndServiceExists(dynamicServerPodName, domainUid, domainNamespace);
+    assertDoesNotThrow(() -> assertTrue(checkClusterReplicaCountMatches(CLUSTER_1,
+        domainUid, domainNamespace, newReplicaCount)));
+
+    // use clusterStatus.sh to verify scaling results
+    result =  assertDoesNotThrow(() ->
+        executeLifecycleScript(STATUS_CLUSTER_SCRIPT, CLUSTER_LIFECYCLE, CLUSTER_1),
+        String.format("Failed to run %s", STATUS_CLUSTER_SCRIPT));
+
+    String regex = ".*" + CLUSTER_1 + "(\\s+)1(\\s+)5(\\s+)2(\\s+)2(\\s+)2";
+    assertTrue(verifyExecuteResult(result, regex),"The script should scale up the cluster: " + CLUSTER_1);
+    logger.info("Dynamic cluster " + CLUSTER_1 + " scaled successfully.");
+
+    // check managed server from other cluster are not affected
+    logger.info("Check configured managed server pods are not affected");
+    assertDoesNotThrow(() -> assertTrue(checkClusterReplicaCountMatches(CLUSTER_2,
+        domainUid, domainNamespace, replicaCount)));
+    checkPodDoesNotExist(configServerPodName, domainUid, domainNamespace);
+
+    // restore test env
+    logger.info("Stop managed server using the script");
+    executeLifecycleScript(STOP_SERVER_SCRIPT, SERVER_LIFECYCLE, dynamicServerName);
```

Comment:
Can you use the same script scaleCluster.sh to scale down the cluster instead of stopServer.sh script

Reply:
will do

Reply:
changed